### DISCUSSION TITLE
Lesson 4: 3:44:37 - I'm unable to compile with the request... python deploy.py in the terminal

### DISCUSSION THREAD

**Discussion by richwealthyandsuccessful** [smartcontractkit/full-blockchain-solidity-course-py#1563] [Q&A] [answered]
I'm unable to successfully compile with the request `python deploy.py` in the terminal, like Patrick does in the video.
```Python
from solcx import compile_standard, install_solc

with open("./SimpleStorage.sol", "r") as file:
    simple_storage_file = file.read()
    print(simple_storage_file)

install_solc("0.6.0")
compiled_sol = compile_standard(
    {
        "language": "Solidity",
        "sources": {"SimpleStorage.sol": {"content": simple_storage_file}},
        "settings": {
            "outputSelection": {
                "*": {"*": ["abi", "metadata", "evm.bytecode", "evm.sourceMap"]}
            }
        },
    },
    solc_version="0.6.0",
)
print(compiled_sol)
```
Currently using a Google HP ChromeBook with Chrome OS, VSCode, Python 3.9.2 ('venv':venv). 

Here is the error message...

```
Traceback (most recent call last):
  File "/home/marcusdahibbert/demos/web3_py_simple_storage/deploy.py", line 11, in <module>
    compiled_sol = compile_standard(
  File "/home/marcusdahibbert/demos/web3_py_simple_storage/venv/lib/python3.9/site-packages/solcx/main.py", line 394, in compile_standard
    raise SolcError(
solcx.exceptions.SolcError: Only "Solidity" or "Yul" is supported as a language.
> command: `/home/marcusdahibbert/.solcx/solc-v0.6.0 --standard-json`
> return code: `0`
> stdout:
{"errors":[{"component":"general","formattedMessage":"Only \"Solidity\" or \"Yul\" is supported as a language.","message":"Only \"Solidity\" or \"Yul\" is supported as a language.","severity":"error","type":"JSONError"}]}

> stderr:
```
What should I do next to resolve this problem?

**Comment by cromewar** [comment, marked as answer]
Hello there, the compiler is complaining as you were not selected the correct language: Solidity but you actually did, so I have some recommendations for you:

1. Be sure you have saved the file before running it.
2. Have you installed py-solc-x properly?
3. please clone the repo of the course and try to run it, does it work?
4. Try yo look an indentation error, most likely on compile_standard().

**Reply by richwealthyandsuccessful** [reply]
Okay, I'll try these steps and let you know how I get on.

Thanks for your quick reply and advice

**Reply by richwealthyandsuccessful** [reply]
Fantastic, instead of saving with the keyboard shortcuts (ctrl = S), I saved the file from the navigation bar (files>save) and then I ran `python deploy.py` and finally it appears to be working.

Thank you Cromewar.
